src/main/java/com/wojciechbarwinski/demo/epic_board_games_shop/dtos/AddressDTO.java

```diff
@@ -1,7 +1,21 @@
 package com.wojciechbarwinski.demo.epic_board_games_shop.dtos;
 
-public record AddressDTO(String street,
-                         String city,
-                         String zipCode,
-                         String phoneNumber) {
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
+
+public record AddressDTO(
+
+        @NotBlank(message = "Street is required")
+        String street,
+
+        @NotBlank(message = "City is required")
+        String city,
+
+        @NotBlank(message = "ZIP code is required")
+        @Pattern(regexp = "\\d{2}-\\d{3}", message = "Invalid ZIP code format. Expected format is XX-XXX")
+        String zipCode,
+
+        @NotBlank(message = "Phone number is required")
+        @Pattern(regexp = "\\d{9}", message = "Invalid phone number format. Expected format is XXXXXXXXX")
```

Comment:
Technicznie rzecz biorącpowinniśmy jeszcze uwzględnić nr z +48, czy podawane w postaci xxx-xxx-xxx albo xxx xxx xxx, ale już luz - to bardziej jako info na przyszłość ;)

Reply:
wiem, problem mam taki, ze to moze być albo bardziej zaawansowany regex... ale dalej pewnie są formy które nie przejdą a powinny 
String regex = "^\\+?\\d{0,3}[\\s-]?\\d{3}[\\s-]?\\d{3}[\\s-]?\\d{3}$"; może być
albo mamy regex który po prostu sprawdzi czy jest odpowiednia ilosc znakow licząć cokowliek i ostatecznie nie ma on znaczenia -.-
sam nie wiem co jest lepsze

Reply:
Na potrzeby nauki ten w zupełności wystarczy ;) Ewentualnie możesz zrobić zadanko w Trello i zostawić komentarz dla tego, kto będzie czytał ten kod na rekru, żeby wiedział, że Ty wiesz, że to nie jest idealny regexp :D

Resolvuj komentarz po odczytaniu